src/logic/taskcluster.ts

```diff
@@ -0,0 +1,67 @@
+// This file contains logic for the Taskcluster Third-Party Login
+
+import { UserCredentials } from '../types/types';
+
+export const prodTaskclusterUrl = 'https://firefox-ci-tc.services.mozilla.com';
+export const stagingTaskclusterUrl =
+  'https://stage.taskcluster.nonprod.cloudops.mozgcp.net';
+export const stagingClientId = 'perfcompare-staging-client';
+export const tcClientIdMap: Record<string, string> = {
+  'https://perf.compare': 'production',
+  'https://beta--mozilla-perfcompare.netlify.app': 'beta',
+  'http://localhost:3000': 'localhost-3000',
+};
```

Comment:
optional nit:
Instead of doing this small concatenation later:
```
    clientId: `perfcompare-${tcClientIdMap[window.location.origin]}-client`,
```
I'd just specify the full values directly here.

Advantages:
* more clarity about all the cliend_ids the project use
* easier to search from a client_id value
* more straightforward code

Disavantages:
* longer to type and to write